src/hostnet/lib/slirp.ml

```diff
@@ -114,17 +114,27 @@ let connect x peer_ip local_ip =
                         Tcpip_stack.IPV4.writev (Tcpip_stack.ipv4 s) ethernet_ip_hdr [ reply ];
                       end else begin
                         let payload = Cstruct.sub udp Wire_structs.sizeof_udp (length - Wire_structs.sizeof_udp) in
+                        let for_us = Ipaddr.V4.compare dst local_ip == 0 in
                         (* We handle DNS on port 53 ourselves, see [listen_udpv4] above *)
                         (* ... but if it's going to an external IP then we treat it like all other
                            UDP and NAT it *)
-                        if Ipaddr.V4.compare dst local_ip != 0 then begin
+                        if (not for_us) then begin
                           Log.debug (fun f -> f "UDP %s:%d -> %s:%d len %d"
                                        (Ipaddr.V4.to_string src) src_port
                                        (Ipaddr.V4.to_string dst) dst_port
                                        length
                                    );
                           let reply buf = Tcpip_stack.UDPV4.writev ~source_ip:dst ~source_port:dst_port ~dest_ip:src ~dest_port:src_port (Tcpip_stack.udpv4 s) [ buf ] in
                           Socket.Datagram.input ~reply ~src:(src, src_port) ~dst:(dst, dst_port) ~payload
+                        end
+                        else if for_us && dst_port == 123 then begin
```

Comment:
And `=` rather than `==` here? I think in this case it doesn't matter but the intent is confusing.